src/views/projects/SourceBrowser/FileDiff.svelte

```diff
@@ -16,6 +17,48 @@
   export let projectId: string;
 
   let collapsed = false;
+  let selection: Selection | undefined = undefined;
+
+  onMount(async () => {
+    window.addEventListener("hashchange", updateSelection);
+  });
+
+  onDestroy(() => {
+    window.removeEventListener("hashchange", updateSelection);
```

Comment:
not sure I understand, you want me to use only `onMount`? I only borrowed this piece of code from **src/views/projects/Blob.svelte**.

Reply:
@sebastinez no, since the onMount is async

Reply:
@hecatia-elegua you are right with the async part, but I don't think we need the `onMount` function to be async, since we don't await anything.

@xphoniex I leave it to you if you want to return from `onMount` or use the additional `onDestroy`